internal/testrunner/runners/system/servicedeployer/custom_agent.go

```diff
@@ -4,33 +4,245 @@
+	}
+
+	// Shortcut to first port for convenience
+	if len(outCtxt.Ports) > 0 {
+		outCtxt.Port = outCtxt.Ports[0]
+	}
+
+	outCtxt.Agent.Host.NamePrefix = inCtxt.Name
+	service.ctxt = outCtxt
+	return &service, nil
+}
+
+// Signal sends a signal to the service.
+func (s *deployedCustomAgent) Signal(signal string) error {
+	p, err := compose.NewProject(s.project, s.ymlPaths...)
+	if err != nil {
+		return errors.Wrap(err, "could not create Docker Compose project for service")
+	}
+
+	opts := compose.CommandOptions{ExtraArgs: []string{"-s", signal}}
+	if s.ctxt.Name != "" {
+		opts.Services = append(opts.Services, s.ctxt.Name)
+	}
+
+	return errors.Wrapf(p.Kill(opts), "could not send %q signal", signal)
+}
+
+// TearDown tears down the service.
+func (s *deployedCustomAgent) TearDown() error {
+	logger.Debugf("tearing down service using Docker Compose runner")
+	defer func() {
+		err := files.RemoveContent(s.ctxt.Logs.Folder.Local)
+		if err != nil {
+			logger.Errorf("could not remove the service logs (path: %s)", s.ctxt.Logs.Folder.Local)
+		}
+
+		if err := os.Remove(s.cfg); err != nil {
+			logger.Errorf("cleaning up tmp file (path: %s): %v", s.cfg, err)
+		}
+	}()
+
+	p, err := compose.NewProject(s.project, s.ymlPaths...)
+	if err != nil {
+		return errors.Wrap(err, "could not create Docker Compose project for service")
+	}
+
+	opts := compose.CommandOptions{
+		Env: s.env,
+	}
+	processServiceContainerLogs(p, opts, s.ctxt.Name)
+
+	if err := p.Down(compose.CommandOptions{
+		Env:       s.env,
+		ExtraArgs: []string{"--volumes"}, // Remove associated volumes.
+	}); err != nil {
+		return errors.Wrap(err, "could not shut down service using Docker Compose")
+	}
+	return nil
+}
+
+// Context returns the current context for the service.
+func (s *deployedCustomAgent) Context() ServiceContext {
+	return s.ctxt
+}
+
+// SetContext sets the current context for the service.
+func (s *deployedCustomAgent) SetContext(ctxt ServiceContext) error {
+	s.ctxt = ctxt
+	return nil
+}
+
+func createCustomAgentYaml(cfgPath string) (string, error) {
+	bc := struct {
+		Version  string                            `yaml:"version"`
+		Services map[string]map[string]interface{} `yaml:"services"`
+	}{}
+	if err := yaml.Unmarshal([]byte(customAgentYml), &bc); err != nil {
+		return "", errors.Wrap(err, "unmarshal base custom-agent config")
+	}
+
+	cb, err := os.ReadFile(cfgPath)
+	if err != nil {
+		return "", errors.Wrap(err, "open custom-agent config")
+	}
+
+	cv := map[string]interface{}{}
+	if err := yaml.Unmarshal(cb, &cv); err != nil {
+		return "", errors.Wrap(err, "unmarshal custom-agent config")
+	}
+
+	for k, v := range cv {
+		bc.Services["custom-agent"][k] = v
+	}
+
+	b, err := yaml.Marshal(bc)
+	if err != nil {
+		return "", errors.Wrap(err, "marshal custom-agent config")
+	}
+
+	dir, err := os.MkdirTemp("", "elastic-package")
+	if err != nil {
+		return "", errors.Wrap(err, "create tmp dir")
+	}
+
+	tf, err := os.CreateTemp(dir, "custom-agent")
```

Comment:
`dir` is never removed? Maybe you can create the config directly on tmp.
```suggestion
	tf, err := os.CreateTemp(os.TempDir(), "custom-agent")
```

Reply:
Makes sense, will just create in on tmp. 👍